pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -313,4 +320,13 @@ def get_pixeldata(ds):
     if should_change_PhotometricInterpretation_to_RGB(ds):
         ds.PhotometricInterpretation = "RGB"
 
+    if not read_only and ds.BitsAllocated > 1:
+        return arr.copy()
+    elif read_only and ds.BitsAllocated == 1:
+        warnings.warn(
```

Comment:
I'm not sure if this warning is needed - the user should not care about the type of image, and just set the flag if he doesn't want to write to the array. The flag should only ensure that you can write to it if it is not set IMO.

Reply:
Hmm, it is probably overkill since its already mentioned in the docstring. OK, I'll remove the warning.